pkg/psmdb/mongo/mongo.go

```diff
@@ -445,6 +445,36 @@ func (m *ConfigMembers) RemoveOld(compareWith ConfigMembers) (changes bool) {
 	return changes
 }
 
+// ExternalNodesChanged checks if votes or priority fields changed for external nodes
+func (m *ConfigMembers) ExternalNodesChanged(compareWith ConfigMembers) (changes bool) {
+	cm := make(map[string]struct {
+		votes    int
+		priority int
+	}, len(compareWith))
+
+	for _, member := range compareWith {
+		_, ok := member.Tags["external"]
+		if !ok {
+			continue
+		}
+		cm[member.Host] = struct {
+			votes    int
+			priority int
+		}{votes: member.Votes, priority: member.Priority}
+	}
+
+	for i := 0; i < len(*m); i++ {
+		member := []ConfigMember(*m)[i]
+		if ext, ok := cm[member.Host]; ok {
+			changes = ext.votes != member.Votes || ext.priority != member.Priority
```

Comment:
1. let's make it more readable
`changes = (ext.votes != member.Votes || ext.priority != member.Priority)`

2. If we `changes` become true on first iteration, we do not "remember" it, and overwrite on the next iteration, which could be the same and assign `false` value to it

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/729/commits/db744b631bde51f6969c4340d944ef36d2ce2b10